apps/ledger-live-mobile/src/components/RootNavigator/BaseNavigator.tsx

```diff
@@ -391,16 +394,20 @@ export default function BaseNavigator() {
             cardStyleInterpolator: CardStyleInterpolators.forVerticalIOS,
           }}
         />
-        <Stack.Screen
-          name={ScreenName.MarketCurrencySelect}
-          component={MarketCurrencySelect}
-          options={{
-            title: t("market.filters.currency"),
-            headerLeft: () => null,
-            // FIXME: ONLY ON BOTTOM TABS AND DRAWER NAVIGATION
-            // unmountOnBlur: true,
-          }}
-        />
+        {marketNewArch?.enabled ? (
+          MarketNavigator({ Stack })
```

Comment:
What I want to do is to compose our Navigators. Initially, I tried with your recommendation:
`<MarketNavigator Stack={Stack}/>`, but if I do so I have the following error:
<img width="328" alt="image" src="https://github.com/LedgerHQ/ledger-live/assets/31533861/de138638-2703-4587-a6e5-9c730c3592c2">

This is well described here: https://github.com/react-navigation/react-navigation/issues/9578

I want to compose our Navigators for two reasons:
- Keep our Navigators close to our features. Each feature should be able to register the screens without having to do it in a common file as we have today.
- Ease Integration tests, as you will test User Stories, and most of the time that means navigation between multiple screens of a feature. If you have a Navigator for your feature you can render it directly without rebuilding it for each test. (This was not so easy for the searchACoin Integration test as we have a specific use case with the feature injected in a TabNavigator)

The trick I found is to call the function directly to avoid the error, it works and I have not found any impact (Maybe I have missed something?)
In conclusion, `MarketNavigator` is a React function component so I suggest that we stick with PascalCase.